pkg/module_manager/module.go

```diff
@@ -885,61 +886,30 @@ func (m *Module) runEnabledScript(precedingEnabledModules []string, logLabels ma
 	return moduleEnabled, nil
 }
 
-var ValidModuleNameRe = regexp.MustCompile(`^[0-9][0-9][0-9]-(.*)$`)
-
-func SearchModules(modulesDir string) (modules []*Module, err error) {
-	if modulesDir == "" {
-		log.Warnf("Modules directory path is empty! No modules to load.")
-		return nil, nil
-	}
-
-	files, err := os.ReadDir(modulesDir) // returns a list of modules sorted by filename
-	if err != nil {
-		return nil, fmt.Errorf("list modules directory '%s': %s", modulesDir, err)
+// RegisterModules load all available modules from modules directory.
+func (mm *moduleManager) RegisterModules() error {
+	if mm.ModulesDir == "" {
+		log.Warnf("Empty modules directory is passed! No modules to load.")
+		return nil
 	}
 
-	badModulesDirs := make([]string, 0)
-	modules = make([]*Module, 0)
-
-	for _, file := range files {
-		if !file.IsDir() {
-			continue
-		}
-		matchRes := ValidModuleNameRe.FindStringSubmatch(file.Name())
-		if matchRes != nil {
-			moduleName := matchRes[1]
-			modulePath := filepath.Join(modulesDir, file.Name())
-			module := NewModule(moduleName, modulePath)
-			modules = append(modules, module)
-		} else {
-			badModulesDirs = append(badModulesDirs, filepath.Join(modulesDir, file.Name()))
-		}
-	}
+	log.Debug("Search and register modules")
 
-	if len(badModulesDirs) > 0 {
-		return nil, fmt.Errorf("modules directory contains directories not matched ValidModuleRegex '%s': %s", ValidModuleNameRe, strings.Join(badModulesDirs, ", "))
+	// load global and modules common static values from modules/values.yaml
+	commonStaticValues, err := LoadCommonStaticValues(mm.ModulesDir)
+	if err != nil {
+		return fmt.Errorf("load common values for modules: %s", err)
 	}
-
-	return
-}
-
-// RegisterModules load all available modules from modules directory
-// FIXME: Only 000-name modules are loaded, allow non-prefixed modules.
-func (mm *moduleManager) RegisterModules() error {
-	log.Debug("Search and register modules")
+	mm.commonStaticValues = commonStaticValues
 
 	modules, err := SearchModules(mm.ModulesDir)
 	if err != nil {
 		return err
 	}
```

Comment:
So, if name != restoredName, we'll get an error. Is that an intended case?

Reply:
Yes, kebab-case to camelCase is a conversion to get a values key and camelCase to kebab-case is a conversion when ConfigMap is loaded.